src/hooks/useWalletsHiddenBalances.ts

```diff
@@ -0,0 +1,113 @@
+import { AllRainbowWallets } from '@/model/wallet';
+import { useMemo, useState, useEffect, useCallback } from 'react';
+import { Address } from 'viem';
+import useAccountSettings from './useAccountSettings';
+import { useConnectedToHardhatStore } from '@/state/connectedToHardhat';
+import { NativeCurrencyKey } from '@/entities/nativeCurrencyTypes';
+import { userAssetsStore } from '@/state/assets/userAssets';
+import { queryClient } from '@/react-query';
+import { userAssetsQueryKey, UserAssetsResult } from '@/resources/assets/UserAssetsQuery';
+import { convertAmountAndPriceToNativeDisplay, add, isEqual } from '@/helpers/utilities';
+import { isEqual as _isEqual } from 'lodash';
+
+export type WalletBalanceResult = {
+  hiddenBalances: Record<Address, string>;
+};
+
+const getHiddenAssetBalance = ({
+  address,
+  nativeCurrency,
+  connectedToHardhat,
+}: {
+  address: Address;
+  nativeCurrency: NativeCurrencyKey;
+  connectedToHardhat: boolean;
+}) => {
+  const hiddenAssetIds = userAssetsStore.getState(address).getHiddenAssetsIds();
+  const assetData = queryClient.getQueryData<UserAssetsResult>(
+    userAssetsQueryKey({ address, currency: nativeCurrency, connectedToHardhat })
+  );
+
+  const balance = hiddenAssetIds.reduce((acc, uniqueId) => {
+    const asset = assetData?.[uniqueId];
+    let sum = acc;
+    if (asset) {
+      const priceUnit = asset.price?.value ?? 0;
+      const nativeDisplay = convertAmountAndPriceToNativeDisplay(asset?.balance?.amount ?? 0, priceUnit, nativeCurrency);
+      sum = add(sum, nativeDisplay.amount);
+    }
+    return sum;
```

Comment:
do we need `convertAmountAndPriceToNativeDisplay` here?

```suggestion
    const asset = assetData?.[uniqueId];
    if (!asset) return acc
    const assetNativeBalance = multiply(asset.price?.value || 0, asset.balance?.amount || 0)
    return add(acc, assetNativeBalance);
```